lib/rucio/core/transfer.py

```diff
@@ -1373,6 +1382,14 @@ def __list_transfer_requests_and_source_replicas(total_workers=0, worker_number=
                                          models.RSEFileAssociation.rse_id == models.Distance.src_rse_id)) \
         .with_hint(models.Distance, "+ index(distances DISTANCES_PK)", 'oracle')
 
+    # if transfertool specified, select only the requests where the source rses are set up for the transfer tool
+    if transfertool:
+        query = query.subquery()
+        query = session.query(query) \
+            .join(models.RSEAttrAssociation, models.RSEAttrAssociation.rse_id == query.c.rse_id) \
+            .filter(models.RSEAttrAssociation.key == 'transfertool',
+                    models.RSEAttrAssociation.value.like(transfertool))
```

Comment:
Shouldnt this be a LIKE against `%transfertool%`? Since an RSE could have the tag `tranfertool1,transfertool2,transfertool3`?

Reply:
You are correct.  I changed the check to '%' + transfertool + '%' so that the attribute can have multiple transfer tools.  I tested and transfers were successful.  However, does this raise risk of injection?  Not sure how/if sqlalchemy sanitizes but just a thought.

Reply:
It's not a huge risk here, since the `transfertool` setting comes from the config / command line parameters on the node. With that level of access you can as well read the database credentials from the config.